app/routes/($locale).cart.tsx

```diff
@@ -33,6 +33,57 @@ export interface CartReturn {
   cart: Cart;
 }
 
+// Helper function to ensure Coveo client ID and tracking ID are set as cart attributes
+async function setCoveoConfigAttributes(
+  context: any,
+  request: Request,
+  cartResult: CartQueryDataReturn,
+) {
+  const {cart: cartHandler} = context;
+  const cart = cartResult.cart;
+
+  if (!cart) return cartResult;
+
+  const attributes = cart.attributes;
+  const attributesToUpdate = [];
+
+  const navigatorProvider = new ServerSideNavigatorContextProvider(request);
+
+  const clientId = navigatorProvider.clientId;
+  const trackingId = engineConfig.configuration.analytics.trackingId;
+  const organizationId = engineConfig.configuration.organizationId;
+  const accessToken = engineConfig.configuration.accessToken;
+
+  const hasAttr = (key: string) => attributes.some((attr) => attr.key === key);
```

Comment:
I tried running locally and adding a product but ran into a 500 here I think.
<img width="1516" height="419" alt="Screenshot 2025-09-09 at 6 48 59 AM" src="https://github.com/user-attachments/assets/08450cab-0071-422e-8db8-bf532667cceb" />
<img width="1209" height="1043" alt="Screenshot 2025-09-09 at 6 49 13 AM" src="https://github.com/user-attachments/assets/f1b7c1e1-dd6d-4ee1-8ac3-ea53a8a071ab" />